test_conformance/spirv_new/test_op_spec_constant.cpp

```diff
@@ -0,0 +1,182 @@
+                              device_results.data(), 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to copy from numbers_buffer");
+
+    for (int i = 0; i < num; i++)
+    {
+        if (device_results[i] == references[i])
+        {
+            log_error("Values match but should not at location %d expected %d "
+                      "obtained %d\n",
+                      i, references[i], device_results[i]);
+            return TEST_FAIL;
+        }
+    }
+
+    err = clSetProgramSpecializationConstant(prog, 101, sizeof(T),
+                                             &spec_constant_value);
+    SPIRV_CHECK_ERROR(err, "Failed to run clSetProgramSpecializationConstant");
+
+    err = clBuildProgram(prog, 1, &deviceID, NULL, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to build program");
+
+    kernel = clCreateKernel(prog, "spec_const_kernel", &err);
+    SPIRV_CHECK_ERROR(err, "Failed to create kernel");
+
+    err = clSetKernelArg(kernel, 0, sizeof(clMemWrapper), &numbers_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument inputs_buffer");
+    err = clSetKernelArg(kernel, 1, sizeof(clMemWrapper), &spec_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument spec_constant_value");
+
+    err = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, &global, NULL, 0, NULL,
+                                 NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to enqueue cl kernel");
+    clFinish(queue);
+
+    std::fill(device_results.begin(), device_results.end(), 0);
+    err = clEnqueueReadBuffer(queue, numbers_buffer, CL_TRUE, 0, bytes,
+                              device_results.data(), 0, NULL, NULL);
+    SPIRV_CHECK_ERROR(err, "Failed to copy from numbers_buffer");
+
+    for (int i = 0; i < num; i++)
+    {
+        if (device_results[i] != references[i])
+        {
+            log_error(
+                "Values do not match in index %d expected %d obtained %d\n", i,
+                references[i], device_results[i]);
+            return TEST_FAIL;
+        }
+    }
+    return TEST_PASS;
+}
+
+
+#define TEST_SPEC_CONSTANT(NAME, type, value, spec_constant_value)             \
+    TEST_SPIRV_FUNC_VERSION(op_spec_constant_##NAME##_simple, Version(2, 2))   \
+    {                                                                          \
+        std::vector<type> inputs(1024, (type)value);                           \
+        std::vector<type> references(1024,                                     \
+                                     (type)value + (type)spec_constant_value); \
+        return test_spec_constant(deviceID, context, queue,                    \
+                                  "op_spec_constant_" #NAME "_simple", inputs, \
+                                  references, (type)spec_constant_value);      \
+    }
+
+// type name, type, value init, spec constant value
+TEST_SPEC_CONSTANT(int, cl_int, 20, -11)
+TEST_SPEC_CONSTANT(uint, cl_uint, 25, 43)
```

Comment:
It is of no real value to have separate signed and unsigned versions of the tests for each size of integer type, because signedness is not visible within SPIR-V, nor is it visible to the `clSetProgramSpecializationConstant()` API function. Please could you remove _either_ the signed tests _or_ the unsigned tests?

Reply:
Note: "_each size_ of integer type". Could you please do this for the other SPIR-V Integer types, i.e. 8-bit (uchar), 16-bit (ushort) and 32-bit (ulong), too? Thanks.